TemporaryProjects.14.0/StartPageExtender.cs

```diff
@@ -5,27 +5,33 @@
 using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Task = System.Threading.Tasks.Task;
 
 namespace TemporaryProjects
 {
     internal sealed class StartPageExtender : IVsWindowFrameEvents
     {
         private readonly DTE dte;
+        private const string CookieName = "TemporaryProjects.StartPageExtender.Cookie";
 
-        public static async Task InitializeAsync(AsyncPackage package)
+        // You can test this in the current Visual Studio instance like this:
+        // https://github.com/jcansdale/TestDriven.Net-Issues/wiki/Test-With...VS-SDK
+        [STAThread]
+        public static void Initialize(IVsUIShell7 vsUIShell, DTE dte)
         {
-            await package.JoinableTaskFactory.SwitchToMainThreadAsync();
+            ThreadHelper.ThrowIfNotOnUIThread();
 
-            var dte = (DTE)await package.GetServiceAsync(typeof(DTE));
-            if (dte.Version == "14.0")
+            if (Cookie is uint cookie)
             {
-                // Doesn't work with Visual Studio 2015
-                return;
+                vsUIShell.UnadviseWindowFrameEvents(cookie);
```

Comment:
Can you please help me understand why this is necessary? Can Initialize ever get called twice? And if so, why do we want to unsubscribe from the events? Thanks!

Reply:
> And if so, why do we want to unsubscribe from the events?

I see, because we resubscribe again.

Reply:
It's only necessary if you want to run it more than once inside the current instance of Visual Studio. I thought this might be useful if they've changed the Start Page in Visual Studio 2019 or if someone wanted to have a go at making it work with Visual Studio 2015.

I find this useful when I want to iterate quickly without having to build and install a .vsix inside 
 new instance of Visual Studio.

Does that make sense? 😄

Reply:
> I see, because we resubscribe again. But why not just keep the original subscription instead of recreating it? If Initialize has been called before, we could just skip doing any work. Would that not work?

I assumed (perhaps incorrectly) that I'd be able to update the code, close the Start Page, run `Initialize` and open the Start Page again. I'd quickly see if my new code was working as hoped.

Reply:
That makes sense although did you mean to say `AdviseWindowFrameEvents` instead of `UnadviseWindowFrameEvents` here?

> Without the unadvise, code I'd end up with this after running UnadviseWindowFrameEvents multiple times: